packages/orchestration/test/test-chain-name-hub.js

```diff
@@ -0,0 +1,213 @@
+      makeNameHubKit();
+    const { nameHub: traceNames, nameAdmin: traceNamesAdmin } =
+      makeNameHubKit();
+
+    for (const { denom, path, baseDenom } of denoms) {
+      const brandName = denomsToBrandName[baseDenom || denom];
+      await brandNamesAdmin.provideChild(denom);
+      brandNamesAdmin.update(denom, brandName); // FIXME, use ERTP Brand
+
+      await denomNamesAdmin.provideChild(brandName);
+      denomNamesAdmin.update(brandName, denom);
+
+      await traceNamesAdmin.provideChild(denom);
+      traceNamesAdmin.update(denom, harden({ baseDenom, path }));
+      await traceNamesAdmin.provideChild(brandName);
+      traceNamesAdmin.update(brandName, harden({ baseDenom, path }));
+    }
+    child.nameAdmin.update('brand', brandNames);
+    child.nameAdmin.update('denom', denomNames);
+    child.nameAdmin.update('trace', traceNames);
+
+    const { nameHub: connectionNames, nameAdmin: connectionNamesAdmin } =
+      makeNameHubKit();
+    for (const {
+      connectionInfo,
+      transferChannels,
+      chainId: lChainId,
+      lastUpdated,
+    } of connections) {
+      const chainName = chainIdToChainName[lChainId];
+      await connectionNamesAdmin.provideChild(chainName);
+      connectionNamesAdmin.update(
+        chainName,
+        harden({
+          chainId: lChainId,
+          connectionInfo: {
+            ...connectionInfo,
+            ...lastUpdated,
+          },
+          // FIXME, there can be more than one, but typically only one canonical
+          transferChannel: transferChannels[0],
+        }),
+      );
+    }
+    child.nameAdmin.update('connection', connectionNames);
+  }
+  return { chainNames, chainNamesAdmin };
+};
+
+test('chain-name-hub', async t => {
+  const { chainNames } = await makeChainNames();
+
+  const chainId = await chainNames.lookup('osmosis', 'chainId');
+  t.is(chainId, 'osmosis-test');
+  const bondAsset = await chainNames.lookup('osmosis', 'bondAsset');
+  t.deepEqual(bondAsset, { brand: 'XXX_agoricNames_brand', denom: 'uosmo' });
+
+  // lookup up Brand from denom
+  const brand = await chainNames.lookup('osmosis', 'brand', 'uosmo');
+  t.is(brand, 'OSMO', 'FIXME, use real ERTP Brand');
+
+  // lookup denom from alleged brand name
+  const denom = await chainNames.lookup('osmosis', 'denom', 'ATOM');
+  t.regex(denom, /ibc\/27394/);
+
+  // lookup denom trace from denom
+  const traceFromDenom = await chainNames.lookup(
+    'osmosis',
+    'trace',
+    'ibc/E7827844CB818EE9C4DB2C159F1543FF62B26213B44CE8029D5CEFE52F0EE596',
+  );
+  const expectedTrace = {
+    baseDenom: 'ubld',
+    path: 'transfer/channel-1',
+  };
+  t.deepEqual(traceFromDenom, expectedTrace);
+  // lookup denom trace from brand name
+  const traceFromBrandName = await chainNames.lookup('osmosis', 'trace', 'BLD');
+  t.deepEqual(traceFromBrandName, expectedTrace);
+
+  // lookup conneciton info for a chain
+  const connection = await chainNames.lookup('osmosis', 'connection', 'agoric');
```

Comment:
I don't expect oasis to decide what the name "agoric" means.

Reply:
We need some mechanism to reference a particular `Connection` in osmosis's list of `connections`. `chainId: string` was a candidate, but I went with `ChainName` (AllegedChainName in the current revision of the PR , but intended to be `chainName`, like [issuerName](https://github.com/Agoric/agoric-sdk/pull/9323/files#r1591226781))

Reply:
I think one of the main reasons to query the 'connection' namespace is to determine the connectionId (from both the host + remote's perspectives if creating an ICA). The other would be to get channelIds of the `ics20-1` channel for MsgTransfers

Reply:
I suspect I would prefer using reverse-lookup for some of this.

Here's hoping to think this over in more detail shortly.